ops/ruby/devops-toolkit/tasks/service_accounts.rb

```diff
@@ -0,0 +1,61 @@
+require 'open3'
+
+# This class avoids the GcpEnvironmentVisitor because we don't want to create or delete
+# any new SA keys, and we're doing everything with the gcloud command line anyway.
+class ServiceAccounts
+  def initialize(options)
+    envs_json = JSON.load(IO.read(options[:'envs-file']))
+    @environments = envs_json['environments'].map { |env| GcpEnvironmentInfo.new(env) }
+    @logger = options[:logger]
+    @is_dry_run = options[:'dry-run'] ? true : false
+  end
+
+  def list_keys
+    @environments.each do |env|
+      get_keys(env).each do |key|
+        @logger.info(key)
+      end
+    end
+  end
+
+  # This tool does not use the GcpEnvironmentVisitor, as it's specifically cleaning up after it.
+  def delete_all_keys
+    @environments.each do |env|
+      keys = get_keys(env)
+      keys.each do |key|
+        if @is_dry_run == true
+          @logger.info("would delete key #{key} for SA #{env.service_account} in project")
+        else
+          delete_key(key, env)
+        end
+      end
+    end
+  end
+
+  # Get key JSON objects
+  def get_keys(env)
+    list_cmd = %W[gcloud iam service-accounts keys list
```

Comment:
would this delete keys that are in active use by other developers?

Reply:
Yes. It's not for the faint of heart. I don't think there's a way to list keys that only I created.

The only reason this is needed is for debugging when you trap the process but forget to let it resume so it can delete the exact key it created.